src/rust/lib/flexer/src/automata/state.rs

```diff
@@ -0,0 +1,108 @@
+use crate::automata::alphabet::Alphabet;
+use crate::automata::state;
+
+use std::ops::RangeInclusive;
+
+
+
+// =======================
+// == State Of Automata ==
+// =======================
+
+/// Flag for invalid state.
+/// When finite automata gets into invalid state
+/// the input sequence of symbols is rejected.
+pub const INVALID:Id = Id {id:usize::max_value()};
+
+/// Newtype wrapper for finite automata input symbol.
+#[derive(Clone,Copy,Debug,PartialEq,Eq,PartialOrd,Ord,Hash)]
+pub struct Symbol {
+    #[allow(missing_docs)]
+    pub val: u32
+}
+
+/// Newtype wrapper for finite automata state ID.
+#[derive(Clone,Copy,Debug,PartialEq,Eq,PartialOrd,Ord,Hash)]
+pub struct Id {
+    #[allow(missing_docs)]
+    pub id: usize
+}
+
+impl Default for Id {
+    fn default() -> Self {
+        state::INVALID
+    }
+}
```

Comment:
What is the reason we want the `Id` to default to invalid? TBH if I see default in the code it would be surprising to me. I think `Id::invalid()` would be much more clear about the meaning.